debezium-core/src/main/java/io/debezium/transforms/tracing/ActivateOpenTracingSpan.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.transforms.tracing;
+
+import java.util.Map;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.data.Envelope;
+import io.debezium.transforms.SmtManager;
+import io.opentracing.Scope;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.Tracer.SpanBuilder;
+import io.opentracing.propagation.Format;
+import io.opentracing.util.GlobalTracer;
+
+/**
+ * This SMT enables integration with an OpenTracing system.
+ * The SMT creates a OpenTracing span and enriches it with metadata from envelope and source info block.<br/>
+ * It is possible to connect the span to a parent span created by a business application.
+ * The application then needs to export its OpenTracing active span into a database field.
+ * The SMT looks for a predefined field name in the {@code after} block
+ * and when found it extracts the parent span from it.<br>
+ * 
+ * See {@link EventDispatcher} for example of such implementation.
+ * 
+ *
+ * @param <R> the subtype of {@link ConnectRecord} on which this transformation will operate
+ * @author Jiri Pechanec
+ */
+public class ActivateOpenTracingSpan<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ActivateOpenTracingSpan.class);
+
+    private static final String DEFAULT_OPENTRACING_SPAN_VALUE = "opentracingspan";
```

Comment:
So is this what's expected by those Kafka producer/consumer interceptors by default?

Reply:
Tha's unrelated to intereceptros but is Debezium specific.